lib/puppetserver/ca/revoke_action.rb

```diff
@@ -0,0 +1,89 @@
+require 'optparse'
+
+module Puppetserver
+  module Ca
+    class RevokeAction
+      # Actions require a short summary constant
+      SUMMARY = 'Revoke a given certificate'
+      BANNER = <<-BANNER
+Usage:
+  puppetserver ca revoke [--help|--version]
+  puppetserver ca revoke CERTNAME
+
+Description:
+Given a valid certname, instructs the CA to revoke it.
+
+Options:
+BANNER
+
+      # class method parser returns an instance of Ruby's OptionParser
+      # (or a custom object that implements summarize() and help().
+      def self.parser(parsed = {})
+        OptionParser.new do |o|
+          o.banner = BANNER
+          o.on('--help', 'Display this revoke specific help output') do |help|
+            parsed['help'] = true
+          end
+          o.on('--version', 'Output the version') do |v|
+            parsed['version'] = true
```

Comment:
This should probably go away. I don't think `--version` in the actions is necessary nor makes a bunch of sense.

Reply:
We might need something like https://github.com/puppetlabs/puppetserver-ca-cli/commit/4a2aa6e2f81454338bfa3a6042a425e959557ea6 though (if we don't define `--help` and `--version` on a parser and it runs across them it will by default raise `SystemExit`. Though I think with the `Cli` class now parses any `--help` and `--version` flags and Does the Right Thing™.

Reply:
I always though it was weird that these we considered flags of the actions in Puppet (at least according to the docs). I would like to see them go away in that context here if we can.

Reply:
Are we going to add the ``--all`` flag?

Reply:
So I'm looking at the docs, and I don't think the revoke command uses the ``--all`` flag. Ignore me.